test/specs/errors.specs.ts

```diff
@@ -33,4 +33,51 @@ describe("Errors", () => {
             "Error: Code: REQUEST_ERROR, HttpCode: 400, Errors: REQUEST_ERROR, REQUEST_ERROR (my no reason field)"
         );
     });
+
+    it("should serialize the ResponseError correctly", async () => {
+        const error = new ResponseError({
+            httpCode: 400,
+            code: ResponseErrorCode.ValidationError,
+            errors: [
+                { reason: ResponseErrorCode.InvalidAmount },
+                { reason: ResponseErrorCode.InvalidFormat, field: "dummy_field" },
+            ],
+        });
+
+        expect(error.toString()).to.equal(
+            "Error: Code: VALIDATION_ERROR, HttpCode: 400, Errors: INVALID_AMOUNT, INVALID_FORMAT (dummy_field)"
+        );
+    });
+
+    it("should serialize an error with `errors` not being an array", async () => {
```

Comment:
probably can shorten this to `should serialize non-array errors`

Reply:
Actually it would be confusing as the `error` object contains the `errors` object/array. Not specifying will make it harder to know which `error` we speak about 🤔

Reply:
Same for the two next comments. I believe that shortening it will introduce some understanding issue as the object is structured in a way which is prone to misunderstanding when describing.

I would feel better being a bit verbose to describe exactly what we want. But I may be missing something here.